mathics/core/parser/parser.py

```diff
@@ -308,13 +308,35 @@ def p_RawLeftAssociation(self, token):
     def p_LeftRowBox(self, token):
         self.consume()
         children = []
-        self.box_depth += 1
-        self.bracket_depth += 1
+        # If this does not happend, it would be because
+        # it was called when a `FormBox` was found.
+        if token.tag == "LeftRowBox":
+            self.box_depth += 1
+            self.bracket_depth += 1
+
         token = self.next()
         while token.tag not in ("RightRowBox", "OtherscriptBox"):
+            if token.tag in "FormBox":
```

Comment:
Should this be in the `while` condition?

Reply:
True!